src/DAL/migration/FullSchema.sql

```diff
@@ -82,8 +82,10 @@ CREATE TABLE "Task"
   "attempts" integer NOT NULL DEFAULT 0,
   "jobId" uuid NOT NULL,
   "resettable" BOOLEAN NOT NULL DEFAULT TRUE,
+  "block_duplication" boolean NOT NULL DEFAULT false,
   CONSTRAINT "PK_task_id" PRIMARY KEY (id), 
-  CONSTRAINT "FK_task_job_id" FOREIGN KEY ("jobId") REFERENCES "Job" (id) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION
+  CONSTRAINT "FK_task_job_id" FOREIGN KEY ("jobId") REFERENCES "Job" (id) MATCH SIMPLE ON UPDATE NO ACTION ON DELETE NO ACTION,
+  CONSTRAINT "UQ_uniqueness_on_job_and_type" EXCLUDE ("type" with =, "jobId" with =) WHERE ("block_duplication" = true)
```

Comment:
Why didn't you use typeORM "@Exclusion"?
https://github.com/typeorm/typeorm/blob/78df84c732ce085caf4b5ccf37477ef93b38f4d0/docs/decorator-reference.md#unique

Reply:
the typeORM decorator are mostly for readability as we don't use it to create the DB schema